jetson/percep/pcl.cpp

```diff
@@ -185,177 +185,81 @@ void PCL::FindInterestPoints(std::vector<pcl::PointIndices> &cluster_indices,
-            leftAngle = newAngle;
-            break;
-        }
-            
-    }
+    //Find Clear left path
+    leftAngle = getAngle(10, 0, interest_points, viewer, obstacles);
 
     //Reset global variables
     obstacles = {0, 0};
     obstacles.at(0) = centerObstacles.at(0);
     obstacles.at(1) = centerObstacles.at(1);
-    newAngle = 0;
-    newSlope = 0;
-    
-    int i = 0; //Sets max iterations
-    //If have to turn too far left check right
-    while(newAngle < 70 && newSlope >= 0 && i < 10){
-        i++;
-       
-       //Declare Stuff
-        double x = pt_cloud_ptr->points[obstacles.at(1)].x;
-        double z = pt_cloud_ptr->points[obstacles.at(1)].z;
-        double zOffset = 0;
-        double xOffset = 0;
-        
-        if(x >= 0){
-            //Calculate angle from X axis, slope of line, and length of line to left obstacle
-            double slope = x/z;
-            double length = sqrt(x*x+z*z);
-            double angleOffset = acos(z/length)*180/PI;
-            
-            //Calculate angle from X axis and length of line that is half rover distance from obstacle
-            double interior = asin((HALF_ROVER+buffer)/length)*180/PI;
-            double adjacent = 90-fabs(interior)-fabs(angleOffset);
-            double compliment = 90-fabs(interior);
-            double key = 180-adjacent-compliment-fabs(interior);
-            zOffset = sin(key*PI/180)*(HALF_ROVER+buffer);
-            xOffset = cos(key*PI/180)*(HALF_ROVER+buffer);
-            }
 
-        else{
-            //Calculate angle from X axis, slope of line, and length of line to left obstacle
-            double slope = x/z;
-            double length = sqrt(x*x+z*z);
-            double angleOffset = acos(z/length)*180/PI;
-
-            //Calculate angle new point will be from X axis
-            double interior = asin((HALF_ROVER+buffer)/length)*180/PI;
-            double key = 90-(fabs(interior)-fabs(angleOffset));
-            xOffset = sin(key*PI/180)*(HALF_ROVER+buffer);
-            zOffset = cos(key*PI/180)*(HALF_ROVER+buffer);
-        }
-        
-        //Find points on line of new path
-        double newPathz = z-fabs(zOffset);
-        double newPathx = x+fabs(xOffset);
-        newSlope = newPathx/newPathz;
-        newAngle = atan(newPathx/newPathz)*180/PI;
-
-        //Create left and right lines based on new path
-        double shiftX = HALF_ROVER/sin((90-newAngle)*PI/180);
-
-        compareLine leftLine (newSlope,-shiftX);
-        compareLine rightLine (newSlope, shiftX);
-
-        obstacles.clear();
-        if(CheckPath(interest_points, viewer, obstacles, leftLine, rightLine)){
-            #if PERCEPTION_DEBUG
-            std::cout<<"!!!!!!!!!!!!!!!!!!!!!!!!!!!!!FOUND NEW PATH AT: "<<newAngle<<std::endl;
-            #endif
-            rightAngle =  newAngle;
-            break;
-        }
-
-    }
+    //Find clear right path
+    rightAngle = getAngle(10, 1, interest_points, viewer, obstacles);
```

Comment:
rightAngle? Sounds like this value should be 90 degrees. Can we make the name more clear? To the right of what?

What are 10 and 1? These should be config variables it seems. That or global/class level constants is my understanding.